tests/components/lacrosse_view/test_sensor.py

```diff
@@ -71,3 +75,31 @@ async def test_field_not_supported(hass: HomeAssistant, caplog) -> None:
     assert entries[0].state == ConfigEntryState.LOADED
     assert hass.states.get("sensor.test_some_unsupported_field") is None
     assert "Unsupported sensor field" in caplog.text
+
+
+async def test_field_types(hass: HomeAssistant) -> None:
+    """Test the different data types for fields."""
+    config_entry = MockConfigEntry(domain=DOMAIN, data=MOCK_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+
+    with patch("lacrosse_view.LaCrosse.login", return_value=True), patch(
+        "lacrosse_view.LaCrosse.get_sensors",
+        return_value=[
+            TEST_FLOAT_SENSOR,
+            TEST_STRING_SENSOR,
+            TEST_ALREADY_FLOAT_SENSOR,
+            TEST_ALREADY_INT_SENSOR,
+        ],
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert hass.data[DOMAIN]
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert entries
+    assert len(entries) == 1
+    assert entries[0].state == ConfigEntryState.LOADED
+    assert hass.states.get("sensor.test_temperature")
+    assert hass.states.get("sensor.test_wet_dry")
+    assert hass.states.get("sensor.test_wind_speed")
+    assert hass.states.get("sensor.test_heat_index")
```

Comment:
These should check not only that the expected sensor exists, but that it also delivers the expected value (or at least its type, considering this PR extended `get_value` to perform exactly that).